Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -0,0 +1,910 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <TerrainRenderer/TerrainDetailMaterialManager.h>
+#include <TerrainRenderer/Components/TerrainSurfaceMaterialsListComponent.h>
+
+#include <AzCore/Console/Console.h>
+
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+
+#include <SurfaceData/SurfaceDataSystemRequestBus.h>
+
+namespace Terrain
+{
+    namespace
+    {
+        [[maybe_unused]] const char* TerrainDetailMaterialManagerName = "TerrainDetailMaterialManager";
+        const char* TerrainDetailChars = "TerrainDetail";
+    }
+    
+    namespace DetailMaterialInputs
+    {
+        static const char* const BaseColorColor("baseColor.color");
+        static const char* const BaseColorMap("baseColor.textureMap");
+        static const char* const BaseColorUseTexture("baseColor.useTexture");
+        static const char* const BaseColorFactor("baseColor.factor");
+        static const char* const BaseColorBlendMode("baseColor.textureBlendMode");
+        static const char* const MetallicMap("metallic.textureMap");
+        static const char* const MetallicUseTexture("metallic.useTexture");
+        static const char* const MetallicFactor("metallic.factor");
+        static const char* const RoughnessMap("roughness.textureMap");
+        static const char* const RoughnessUseTexture("roughness.useTexture");
+        static const char* const RoughnessFactor("roughness.factor");
+        static const char* const RoughnessLowerBound("roughness.lowerBound");
+        static const char* const RoughnessUpperBound("roughness.upperBound");
+        static const char* const SpecularF0Map("specularF0.textureMap");
+        static const char* const SpecularF0UseTexture("specularF0.useTexture");
+        static const char* const SpecularF0Factor("specularF0.factor");
+        static const char* const NormalMap("normal.textureMap");
+        static const char* const NormalUseTexture("normal.useTexture");
+        static const char* const NormalFactor("normal.factor");
+        static const char* const NormalFlipX("normal.flipX");
+        static const char* const NormalFlipY("normal.flipY");
+        static const char* const DiffuseOcclusionMap("occlusion.diffuseTextureMap");
+        static const char* const DiffuseOcclusionUseTexture("occlusion.diffuseUseTexture");
+        static const char* const DiffuseOcclusionFactor("occlusion.diffuseFactor");
```

Comment:
Nit: For consistency, seems like these should be "OcclusionMap", "OcclusionUseTexture" and "OcclusionFactor"

Reply:
There's both specular and diffuse occlusion, so saying DiffuseOcclusionMap instead of OcclusionMap is clearer. Notice the mateiral properties also say "occlusion.diffuseTextureMap" not "occlusion.textureMap" like most properties.